crates/vm2/src/heap.rs

```diff
@@ -217,20 +217,21 @@ impl Heaps {
     }
 
     fn allocate_inner(&mut self, memory: &[u8]) -> HeapId {
-        let id = HeapId::from_u32_unchecked(self.heaps.len() as u32);
-        self.heaps
+        let id = u32::try_from(self.inner.len()).expect("heap ID overflow");
```

Comment:
Allocating 2^32 heaps with 2^32 gas would be quite a feat. Also, we'd have a memory use problem.

Especially once we start reusing heap ids, this is pretty much impossible. Do we start reusing heap ids, tough? It might be a bit confusing for code that keeps a HeapId for too long but that shouldn't be happening much.

Reply:
I'd consider this a theoretical concern, but for correctness, using `try_from` + `expect` is preferable. IIUC, it doesn't influence performance that much.